netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -136,12 +136,14 @@ public static ByteBufAllocator getByteBufAllocator(boolean forceHeapBuffer) {
     if (Boolean.parseBoolean(
             System.getProperty("io.grpc.netty.useCustomAllocator", "true"))) {
       boolean defaultPreferDirect = PooledByteBufAllocator.defaultPreferDirect();
-      logger.log(
-          Level.FINE,
-          String.format(
-              "Using custom allocator: forceHeapBuffer=%s, defaultPreferDirect=%s",
-              forceHeapBuffer,
-              defaultPreferDirect));
+      if (logger.isLoggable(Level.FINE)) {
```

Comment:
Actually, we shouldn't be using String.format() here. We should use the MessageFormat syntax with the `{0}` stuff. That's what we do elsewhere and I see no reason for this to be different, unless you are concerned of the `new Object[] {forceHeapBuffer, defaultPreferDirect}` allocation. (In which case, you'd use this isLoggable() approach, but still with MessageFormat syntax.)

Reply:
OK